core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -231,9 +240,23 @@ public void run() {
 
   private ClientTransport obtainActiveTransport() {
     ClientTransport savedActiveTransport = activeTransport;
-    if (savedActiveTransport != null) {
+    // If we know there is an active transport and we are not in backoff mode, return quickly.
+    if (savedActiveTransport != null && savedActiveTransport != InactiveTransport.INSTANCE) {
       return savedActiveTransport;
     }
+    if (savedActiveTransport == InactiveTransport.INSTANCE) {
```

Comment:
This condition looks like it should be within the synchronized block and work on `activeTransport`. Otherwise, you can set `activeTransport = null` even if `activeTransport != InactiveTransport.INSTANCE`.